### ISSUE TITLE
Document the current state of which frontend apps use which backend services

### ISSUE THREAD

**Issue by rtluu** [department-of-veterans-affairs/va.gov-team#261]
## User Story
As a backend engineer, I need a full sense of how vets-api is organized so I can clearly articulate how the backend apps are managed (engines, alert triggers, etc.)

## Goal
Identify the main pieces of vets-api and which frontend apps use which backend services

## Acceptance Criteria
- [ ] A mapping of which frontend apps use which backend services

## Timebox
- [ ] 1-2 days

## Definition of Done
- [ ] **Attached to a Milestone** (when will this be completed?)
- [ ] **Attached to an Epic** (what body of work is this a part of?)
- [ ] **Labeled with Team** (`product support`, `analytics-insights`, `operations`, `triage`, `tools-improvements`)
- [ ] **Labeled with Practice Area** (`backend`, `frontend`, `devops`, `design`, `research`, `product`, `ia`, `qa`, `analytics`, `call center`, `research`, `accessibility`, `content`)
- [ ] **Labeled with Type** (`bug`, `request`, `discovery`, `documentation`, etc.)

**Comment by jholton** [comment]
This story was originally created to help with driving out expectations of the platform (VSP) in dealing with external services.

I created [another issue](https://github.com/department-of-veterans-affairs/va.gov-team/issues/326) and opened a corresponding RFC to outline the expectations.

I do think the task described here is worthwhile, and I did start to attempt it.
However it became clear that it is a time intensive endeavor.
For each each Product (e.g. form/application on `vets-website`) there are the obvious external service correlations, there are also prerequisite correlations.  Specifically, most (all?) require `current_user` and form prefill data, which reach out to many services to aggregate user data.
`S3` can also be intermediary dependency. 
The receiving external service can be accessed synchronously or asynchronously.

In short, there are many, many use cases to document, which would be a lot of work for not a lot of immediate benefit.

I think this work should be done as needed (e.g. perhaps when laying out external dependencies for a product for a dashboard).

**Comment by jholton** [comment]
Current frontend products/apps can be listed by visiting `www.va.gov` and opening the dev console and running `settings.applications`.

You can click through to find the “navPath” if you want to open the app.

You can get the current complete list by running `npm run apps` in the `vets-website` repo.

The one you can get from `www.va.gov` has only released apps. The `npm` command list includes some unreleased things.
